props/saber_fett263_buttons.h

```diff
@@ -293,39 +708,754 @@ class SaberFett263Buttons : public virtual PropBase {
+          sound_queue_.Play(SoundToPlay("hundred.wav"));          
+        }
+        // Tens & Ones
+        if (ones == 0) {
+          if ((thousand + hundred) == 0) sound_queue_.Play(SoundToPlay("mzero.wav"));
+        } else {
+          TensValue(ones);
+        }
+        break;
+      default:
+        break;
+    }
+  }
+   
+  enum ClashType {
+    CLASH_NONE,
+    CLASH_CHECK,
+    CLASH_BATTLE_MODE,
+    CLASH_LOCKUP,
+    CLASH_LOCKUP_END,
+#ifdef FETT263_CLASH_STRENGTH_SOUND    
+    CLASH_NORMAL,
+    CLASH_STAB,
+#endif
+  };
+
+#ifdef FETT263_CLASH_STRENGTH_SOUND
+  void SelectClashFile() {
+    if (clash_type_ == CLASH_BATTLE_MODE) {
+      if (SaberBase::GetClashStrength() < saved_gesture_control.clashdetect) {
+        clash_type_ = CLASH_NORMAL;
+      } else {
+        clash_type_ = CLASH_LOCKUP;
+        auto_lockup_on_ = true;
+      }
+    }
+    int files;
+    int file_number;
+    if (saved_gesture_control.maxclash < 8) saved_gesture_control.maxclash = 8;
+    float clash_value = (SaberBase::GetClashStrength() - GetCurrentClashThreshold()) / saved_gesture_control.maxclash;
+    switch (clash_type_) {
+      default:
+        break;
+      case CLASH_NORMAL:
+        if (SFX_clash) {
+          files = SFX_clash.files_found() - 1;
+          file_number = clamp(files * clash_value, 0, files);
+          SFX_clash.Select(file_number);
+        } else {
+          files = SFX_clsh.files_found() - 1;
+          file_number = clamp(files * clash_value, 0, files);
+          SFX_clsh.Select(file_number);
+        }
+        SaberBase::DoClash();
+        break;
+      case CLASH_STAB:
+        if (SFX_stab) {
+          files = SFX_stab.files_found() - 1;
+        } else {
+          if (SFX_clash) {
+            files = SFX_clash.files_found() - 1;
+            file_number = clamp(files * clash_value, 0, files);
+            SFX_clash.Select(file_number);
+          } else {
+            files = SFX_clsh.files_found() - 1;
+            file_number = clamp(files * clash_value, 0, files);
+            SFX_clsh.Select(file_number);
+          }          
+        }
+        SaberBase::DoStab();
+        break;
+      case CLASH_LOCKUP:
+        if (battle_mode_) clash_value = (SaberBase::GetClashStrength() - saved_gesture_control.clashdetect) / saved_gesture_control.maxclash;
+        if (SFX_bgnlock) {
+          files = SFX_bgnlock.files_found() - 1;
+          file_number = clamp(files * clash_value, 0, files);
+        }
+        SaberBase::SetLockup(SaberBase::LOCKUP_NORMAL);
+        SaberBase::DoBeginLockup();
+        break;
+      case CLASH_LOCKUP_END:
+        if (SFX_endlock) {
+          files = SFX_endlock.files_found() - 1;
+          file_number = clamp(files * (fusor.swing_speed() / 600), 0, files);
+        }
+        SaberBase::DoEndLockup();
+        SaberBase::SetLockup(SaberBase::LOCKUP_NONE);
+        break;
+    }
+    clash_type_ = CLASH_NONE;
+  }
+  #endif
+
   void Loop() override {
```

Comment:
Loop() is far too big.
Can we try breaking into functions?

Reply:
broke out the pieces I could to functions

Reply:
Next is the edit mode menu stuff.

Reply:
The edit mode stuff still needs to be broken out of Loop()

Reply:
broke out CheckEvent() and EditColor()